cgmes/cgmes-conversion/src/main/java/com/powsybl/cgmes/conversion/export/TopologyExport.java

```diff
@@ -227,30 +227,33 @@ private static void writeTerminal(String terminalId, String topologicalNode, Str
 
     private static void writeTopologicalNodes(Network network, Set<String> addedTopologicalNodes, String cimNamespace, XMLStreamWriter writer, CgmesExportContext context) throws XMLStreamException {
         writeBusTopologicalNodes(network, addedTopologicalNodes, cimNamespace, writer, context);
-        writeDanglingLineTopologicalNodes(network, cimNamespace, writer, context);
         writeHvdcTopologicalNodes(network, cimNamespace, writer);
-    }
-
-    private static void writeBusTopologicalNodes(Network network, Set<String> addedTopologicalNodes, String cimNamespace, XMLStreamWriter writer, CgmesExportContext context) throws XMLStreamException {
-        for (Bus b : network.getBusBreakerView().getBuses()) {
-            String topologicalNodeId = context.getNamingStrategy().getCgmesId(b);
-            writeTopologicalNode(topologicalNodeId, b.getNameOrId(), b.getVoltageLevel(), addedTopologicalNodes, cimNamespace, writer, context);
-        }
+        // We create topological nodes for boundary side of dangling lines that are not mapped to an external boundary node
+        writeDanglingLineTopologicalNodes(network, cimNamespace, writer, context);
     }
 
     private static void writeDanglingLineTopologicalNodes(Network network, String cimNamespace, XMLStreamWriter writer, CgmesExportContext context) throws XMLStreamException {
         for (DanglingLine dl : network.getDanglingLines()) {
             Optional<String> topologicalNodeId = dl.getAliasFromType(Conversion.CGMES_PREFIX_ALIAS_PROPERTIES + CgmesNames.TOPOLOGICAL_NODE);
-            if (topologicalNodeId.isPresent()) {
+            if (topologicalNodeId.isEmpty()) {
+                // If no information about original boundary has been preserved in the IIDM model,
+                // we will create a new TopologicalNode inside the voltage level of the dangling line network side
                 String baseVoltage = context.getBaseVoltageByNominalVoltage(dl.getTerminal().getVoltageLevel().getNominalV()).getId();
-                // TODO(Luma) It is wrong to consider the container of boundary topological node to be the same of dangling line
-                //  To be fixed in a separate PR (in fact boundary TNs of dangling lines should not be exported)
                 String containerId = context.getNamingStrategy().getCgmesId(dl.getTerminal().getVoltageLevel());
-                writeTopologicalNode(topologicalNodeId.get(), dl.getNameOrId(), containerId, baseVoltage, cimNamespace, writer);
+                String fictTopologicalNodeId = CgmesExportUtil.getUniqueId();
+                dl.addAlias(fictTopologicalNodeId, Conversion.CGMES_PREFIX_ALIAS_PROPERTIES + CgmesNames.TOPOLOGICAL_NODE);
```

Comment:
Can you add "boundary" in the name?

Reply:
added `_Boundary` suffix for the related alias types.